pandas/core/groupby.py

```diff
@@ -598,7 +610,7 @@ def get_group(self, name, obj=None):
         if obj is None:
             obj = self._selected_obj
 
-        inds = self._get_index(name)
+        inds = self._get_index(name, raise_on_missing=True)
```

Comment:
I think I would drop the ``raise_on_missing`` argument and return the default of ``[self.indices.get(name, []) for name in names]`` in ``_get_index``, then for here:

```
if not len(inds):
    raise KeyError(name)
```